process_report/process_report.py

```diff
@@ -1,9 +1,108 @@
 import argparse
 import os
+import sys
 
 import pandas
 
 
+### Invoice field names
+INVOICE_MONTH_FIELD = 'Invoice Month'
+PROJECT_FIELD = 'Project - Allocation'
+MANAGER_PI_FIELD = 'Manager (PI)'
+INSTITUTION_FIELD = 'Institution'
+SU_HOUR_FIELD = 'SU Hours (GBhr or SUhr)'
+SU_TYPE_FIELD = 'SU Type'
+COST_FIELD = 'Cost'
+CREDIT_FIELD = 'Credit'
+CREDIT_CODE_FIELD = 'Credit Code'
+BALANCE_FIELD = 'Balance'
+###
+
+
+def apply_credits_0001(dataframe):
+    credit_code = "0001"
+    pass
+
+
+def apply_credits_0002(dataframe):
+    """Applies the New PI Credit. This credit function expects the 
+    env var `C0002_OLD_PI` to be set, pointing to a txt file containing old PIs"""
+    credit_code = "0002"
+    credit_amount = 1000
+
+    old_pi_list = set()
+    try:
+        with open(os.getenv("C0002_OLD_PI")) as f:
+            for pi in f: old_pi_list.add(pi.strip())
+
+    except Exception:
+        print("Old PI file does not exist, or the C0002_OLD_PI env var is not set")
+        sys.exit(1)
+
+    pi_list = dataframe[MANAGER_PI_FIELD].unique()
+
+    for pi in pi_list:
+        if pi != pi: continue   # NaN check
+        if pi in old_pi_list: continue  # Is the PI an old PI?
+
+        pi_projects = dataframe[dataframe[MANAGER_PI_FIELD] == pi]
+        rem_credit = credit_amount
+        for i, row in pi_projects.iterrows():
+            project_cost = row[COST_FIELD]
+            if project_cost >= rem_credit:
+                dataframe.at[i, CREDIT_FIELD] =  rem_credit
+                dataframe.at[i, CREDIT_CODE_FIELD] = credit_code
+                dataframe.at[i, BALANCE_FIELD] = row[COST_FIELD] - rem_credit
+                break
+            else:
+                dataframe.at[i, CREDIT_FIELD] =  project_cost
+                dataframe.at[i, CREDIT_CODE_FIELD] = credit_code
+                dataframe.at[i, BALANCE_FIELD] = 0
+                rem_credit -= project_cost
+    
+    return dataframe
+
+
+applied_credits = [apply_credits_0002]
+
+
+def get_institution_from_pi(pi_uname):
+    
+    institute_map = {
```

Comment:
@knikolla are we flagging rows that would be useful to add to [this](https://github.com/nerc-project/operations/issues/420) repo once it is ready?